tempesta_fw/str.c

```diff
@@ -318,6 +318,67 @@ tfw_stricmpspn(const TfwStr *s1, const TfwStr *s2, int stop)
 EXPORT_SYMBOL(tfw_stricmpspn);
 
 /**
+ * Generic function for comparing first @len symbols in TfwStrs.
+ * @s1,@s2 may be either plain or compound, but must be longer than @len symbols
+ * Do not use it for duplicate strings, rather call it for each duplicate
+ * substring separately.
+ *
+ * @return true if the strings are equal and false otherwise.
+ */
```

Comment:
It seems both the string functions are unnecessary, see comments above: you can just create temporary compound TfwStr with ":" at the end chunk and use `tfw_stricmpspn()`.